### PR TITLE
Upgrade to opentelemetry-java 1.10.0

### PR DESCRIPTION
**Pull request by jack-berg**
Had to work around some breaking changes:
- Removal of bind api
- Location change of metrics testing assertj classes
- Location change of InMemoryMetricReader
- Removal of GlobalMeterProvider 
- Async instrument change from `.observe(..)` to `.record(..)`

### CODE REVIEW

File: jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/container/ContainerConfigurationHandler.java [hunk 1]
```diff
@@ -6,31 +6,29 @@
...
   public ContainerConfigurationHandler() {
-    initializeMeter(NoopMeter.getInstance());
+    initializeMeter(defaultMeter());
   }
 
   @Override
   public void initializeMeter(Meter meter) {
     meter
-        .upDownCounterBuilder(METRIC_NAME)
+        .upDownCounterBuilder("runtime.jvm.cpu.limit")
```
Comment (kittylyst) [inline jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/container/ContainerConfigurationHandler.java:28]: Why are we inlining this constant? I thought the plan was to standardize on <OPTIONAL_PREFIX>_?METRIC_NAME so it's easy to grep for all of the generated metrics and which class handles them?

File: jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/cpu/OverallCPULoadHandler.java [hunk 2]
```diff
@@ -27,52 +26,36 @@
   private static final String JVM_SYSTEM = "jvmSystem";
   private static final String MACHINE_TOTAL = "machineTotal";
 
-  private static final String METRIC_NAME = "runtime.jvm.cpu.utilization";
-  private static final String DESCRIPTION = "CPU Utilization";
```
Comment (kittylyst) [inline jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/cpu/OverallCPULoadHandler.java:31]: Similar comment as to METRIC_NAME - why are we inlining the description constant?

File: jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/memory/ObjectAllocationOutsideTLABHandler.java [hunk 3]
```diff
@@ -59,19 +57,19 @@ public String getEventName() {
...
-    private final BoundDoubleHistogram boundHistogram;
+    private final DoubleHistogram histogram;
+    private final Attributes attributes;
 
     public PerThreadObjectAllocationOutsideTLABHandler(
         DoubleHistogram histogram, String threadName) {
-      boundHistogram =
-          histogram.bind(Attributes.of(ATTR_THREAD_NAME, threadName, ATTR_ARENA_NAME, MAIN));
+      this.histogram = histogram;
+      this.attributes = Attributes.of(ATTR_THREAD_NAME, threadName, ATTR_ARENA_NAME, "Main");
```
Comment (kittylyst) [inline jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/memory/ObjectAllocationOutsideTLABHandler.java:67]: Should probably dig into / have more discussion around Arena naming - I am not 100% that my understanding of what JFR presents to us is correct.

**Review by kittylyst — COMMENTED** [review]
If we can clear up the query around constant inlining, I think this is good-to-go.

**Review by mateuszrzeszutek — APPROVED** [review]

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java [hunk 4]
```diff
@@ -70,16 +69,24 @@ public GroovyMetricEnvironment(
                 System.setProperty(key, value.toString());
               }
             });
-        // this call will dynamically load the autoconfigure extension
-        // and take care of provider and exporter creation for us based on system properties.
-        GlobalOpenTelemetry.get();
-        meterProvider = (SdkMeterProvider) GlobalMeterProvider.get();
+        // call the autoconfigure extension and take care of provider and exporter creation for us
```
Comment (anuraaga) [inline jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java:72]: Do you feel like migrating the above lines to use `addPropertiesSupplier`?

**Review by anuraaga — APPROVED** [review]

File: jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/container/ContainerConfigurationHandler.java (hunk 1 above)
Comment (jack-berg) [reply jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/container/ContainerConfigurationHandler.java:28]: > I thought the plan was to standardize on <OPTIONAL_PREFIX>_?METRIC_NAME so it's easy to grep for all of the generated metrics and which class handles them?

Ah.. I get why they're not inlined now. That makes sense to me.

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java (hunk 4 above)
Comment (jack-berg) [reply jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java:63]: Good idea 👍

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java [hunk 5]
```diff
@@ -60,26 +60,34 @@ public GroovyMetricEnvironment(
...
+            AutoConfiguredOpenTelemetrySdk.builder()
+                .setResultAsGlobal(false)
+                .addPropertiesSupplier(
+                    () -> {
+                      Map<String, String> properties = new HashMap<>();
+                      // no need for autoconfigure sdk-extension to enable default traces exporter
+                      properties.put("otel.traces.exporter", "none");
+                      // expose config.properties to autoconfigure
+                      config.properties.forEach(
+                          (k, value) -> properties.put(k.toString(), value.toString()));
```
Comment (anuraaga) [inline jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java:75]: Previous code is checking the key starts with otel. or otel.jmx

**Pull request merged**